pkg/util/utils_test.go

```diff
@@ -0,0 +1,151 @@
+/*
+Copyright © 2019 Portworx
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package util
+
+import (
+	"reflect"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+/*
+ To test postivive case of utils.ListContains function.
+ Test if given element is present in the list.
+ Assert if element is not found in the list.
+*/
+func TestListContainsElement(t *testing.T) {
+	// list containg elements
+	elements := []string{"node", "drive", "volume"}
+	matchString := "volume"
+
+	ret := ListContains(elements, matchString)
+	assert.Equal(t, ret, true, "%s entity not found in the list", matchString)
+}
+
+/*
+To test negative case of utils.ListContains function.
+Test if given element is not present in the list.
+Asserts if element is found in the list.
+*/
+func TestListContainsNoElement(t *testing.T) {
+	// list containg elements
+	elements := []string{"node", "drive", "volume"}
+	matchString := "portworx"
+
+	ret := ListContains(elements, matchString)
+	assert.Equal(t, ret, false, "%s entity found in the list", matchString)
+}
+
+/*
+To test positive case of utils.ListHaveMatch.
+Test if given element present in the both the list.
+Assert if none of the elements is not found in both the list.
+*/
+func TestListHaveMatchPresent(t *testing.T) {
+	elements := []string{"node", "drive", "volume", "portworx"}
+	match := []string{"portworx", "osd"}
+
+	ret := ListHaveMatch(elements, match)
+	assert.Equal(t, ret, true, "Elements %s not found in the list", match)
+}
+
+/*
+Test for negative case of utils.ListHaveMatch.
+Tests if the given entity is not present in the list.
+Assert if any one of the elements is found in both the list.
+*/
+func TestListHaveMatchNotPresent(t *testing.T) {
+	elements := []string{"node", "drive", "volume", "portworx"}
+	match := []string{"oci", "osd"}
+
+	ret := ListHaveMatch(elements, match)
+	assert.Equal(t, ret, false, "One of the elements %s found in the list", match)
+}
+
+/*
+Test for positive case of utils.StringMapToCommaString
+Test if the given map can be converted to valid string.
+Assert on conversion failure.
+*/
+func TestStringMapToCommaString(t *testing.T) {
+	elements := map[string]string{
+		"pod":     "portworx",
+		"cluster": "k8s",
+	}
+	expectedResult := "pod=portworx,cluster=k8s"
```

Comment:
The order of the string is not guaranteed by Golang. So this may fail sometimes.

One way is:
map to comma, comma to map.. Should get the same thing back

Another is:
Have pod first or and pod after on two different strings.

Reply:
Done.